### DISCUSSION TITLE
@use-gesture/vanilla Drag Speed or at least the ability to change the offset on your own?

### DISCUSSION THREAD

**Discussion by Blankeos** [pmndrs/use-gesture#615] [General]
First off, thanks for the library. I've been making a **Row List element** with this and so far it's pretty good. Just a little unsatisfying when I scroll really quick and it doesn't move  It depends a lot on **offset** for tracking the current value, **bounds** for constraints, and **rubberband=true** for the crisp animation when it approaches bounds.
I'm building it on Svelte.

I've been reading the docs and I couldn't find anything about increasing the offset speed or at least be able to override the offset values with a multiplier. Even if I do add a multiplier on the offset for setting the coordinates, since it's already hugely different from the real offset value, the bounds won't work anymore.

Is there a way to add more Drag Speed (like when scrolling really fast)?

![demo_drag_routes](https://github.com/pmndrs/use-gesture/assets/38070918/382ae5eb-ff0e-4037-966f-99c828bce924)

```svelte
<script lang="ts">
	import cn from '@/lib/cn';
	import { DragGesture } from '@use-gesture/vanilla';
	import { onMount } from 'svelte';
	import { spring } from 'svelte/motion';

	let coords = spring(
		{ x: 0, y: 0 },
		{
			precision: 1,
			damping: 1
		}
	);
	let screenSize: number;
	let sizeX: number;
	
	function initializeDrag() {
		const el = document.getElementById('row-list-draggable');

		if (!el) return;
		sizeX = el.getBoundingClientRect().width;

		const gesture = new DragGesture(
			el,
			({ active, delta, movement: [mx, my], offset: [ox, oy], direction: [dx, dy] }) => {
				sizeX = el.getBoundingClientRect().width;
				coords.set({
					x: ox,
					y: oy
				});
			},
			{
				axis: 'x',
				bounds: {
					left: screenSize - sizeX,
					right: 0
				},
				rubberband: true
			}
		);
	}

	onMount(initializeDrag);
</script>

<svelte:window bind:innerWidth={screenSize} />
<div
	id="row-list-draggable"
	style:transform="translate({$coords.x}px, {$coords.y}px)"
	class={cn('w-max touch-none')}
>
	<slot />
</div>
```

**Comment by Blankeos** [comment]
Did a little more tinkering around, I guess you could do this as well. It works great!
Using the same multiplier on the bounds. (Only caveat is that when you scroll slowly, it's too fast)
```diff
const gesture = new DragGesture(
	el,
	({ active, delta, movement: [mx, my], offset: [ox, oy], direction: [dx, dy] }) => {
		sizeX = el.getBoundingClientRect().width;
		coords.set({
+			x: ox * 2,
+			y: oy * 2
		});
	},
	{
		axis: 'x',
		bounds: {
+			left: (screenSize - sizeX) / 2,
			right: 0
		},
		rubberband: true
	}
);
```

**Comment by dbismut** [comment]
Hey, is it possible for you to send a small sandbox?

**Reply by Blankeos** [reply]
I've been fiddling with codesandbox to work, it doesn't work the same as the one on my computer. Probably because the svelte version on sandbox is less than `"svelte": "^3.8"`. My version is `"svelte": "^4.0.5"`. 

No worries on this though.

**Reply by dbismut** [reply]
You can upgrade the version on csb as well...
